### ISSUE TITLE
bug: `since` value inconsistent between send/receive during `retrieve`

### ISSUE THREAD

**Issue by techfg** [skuid/skuid-cli#233]
There are several issues with the way that `--since` is handled by the CLI:

1. When `--since` is not specified and there no other filters, `since` is not included in the json sent to the server, however if any other filter is specified (e.g., `--app`) a value of `0001-01-01T00:00:00Z` (the "Zero" time) is marshalled for `--since` even though `--since` was not specified.  While the "net" effect should be the same, the treatment of `--since` is inconsistent based on other factors even when `--since` is not specified.
2. When `--since` is specified, upon retrieval of the retrieve plans, the value of `Since` in each plan returned is inspected and if it is empty string (`""`), the value is updated to the be the value of the `--since` flag.  There are two issues here:
    1. When `--since` is not specified but another filter is, a value for since is sent to the server so in this case, the "sync" of the value in the plan returned and the value that was sent to the server is not performed.
    2. If the server returns a Since value in a plan that does not match the `--since` filter specified, this is an unexpected response as the value should match.  If it doesn't match, that means the server simply did not return the value properly and/or the value it used to filter the response was not the value that was sent in the request.  Either way, if the response value doesn't match the submitted value, this is an error and should not be ignored by simply updating the plan that was returned.  In theory, the response shouldn't even need to be checked for consistency - for example, why check "since" but not "appSpecific" or any other portion of the response?  At some point, the response needs to be able to be trusted - why special case "since"?
3. When "executing" a plan, "since" is always sent in the request to the server, it will contain empty string (`""`) if no filters have been applied, zero time (`0001-01-01T00:00:00Z`) if any other filter but not `since` has been applied, or the value of `--since` if since was applied.
4. When `--since` is specified, the value is converted to UTC and the UTC value sent to the server, a conversion that is completely unnecessary since the value is marshalled in RFC3339 format which will contain the TZ.
5. `since` is stored as a string in each structure that requires it even though since is a `time.Time` and there is no reason to convert to string only making a reverse conversion necessary to use it properly.

To summarize:

1. `since` value are not consistently passed to the server in all situations - when `since` is specified, the value should be the value specified and when not specified, there should be no value sent in any request (or it should be empty/nil at worst in all requests)
2. `since` is unnecessarily converted to a string and formatted outside of json (un)marshalling, deviating from standard usage
3. `since` responses are "corrected" if not what expected instead of an error being thrown

All of the above issues should be addressed so that handling `since` is consistent at every step of the `retrieve` process.
